### PR TITLE
SDIT-899: ✨ Initial stab at modsecurity rules

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: helm_deploy/values-cp-dev.yaml [hunk 1]
```diff
@@ -10,6 +10,28 @@ generic-service:
     enabled: true
     host: prison-api-dev.prison.service.justice.gov.uk
     tlsSecretName: prison-api-cert
+    modsecurity_enabled: true
```
Comment (petergphillips) [inline helm_deploy/values-cp-dev.yaml:13]: unfortunately can't add comments to the snippet below as causes
```
Error: failed to parse values-cp-dev.yaml: error converting YAML to JSON: yaml: line 17: did not find expected key
```
so have to add above

File: helm_deploy/values-cp-dev.yaml [hunk 2]
```diff
@@ -10,6 +10,28 @@ generic-service:
...
+    # Found User-Agent associated with scripting/generic HTTP client - 913101
+    modsecurity_snippet:
+      SecRuleEngine On
+      SecDefaultAction "phase:2,pass,log,tag:github_team=dps-tech"
+      SecRuleRemoveById 920320
+      SecRuleRemoveById 920300
+      SecRuleRemoveById 920440
+      SecRuleRemoveById 942110
+      SecRuleRemoveById 913101
+      SecRuleUpdateTargetById 932150 "!ARGS:/keywords/"
```
Comment (petergphillips) [inline helm_deploy/values-cp-dev.yaml:27]: copied exclusions from https://github.com/ministryofjustice/nomis-api-terraform-azure/blob/main/appgw-v2.tf, looked for anything that isn't auth related

File: helm_deploy/values-cp-dev.yaml [hunk 3]
```diff
@@ -10,6 +10,28 @@ generic-service:
...
+      SecRuleUpdateTargetById 942330 "!ARGS:/keywords/"
+      SecRuleUpdateTargetById 942410 "!ARGS:/outcomeComment/"
+      SecRuleUpdateTargetById 942150 "!ARGS:/outcomeComment/"
+      SecRuleUpdateTargetById 942130 "!ARGS:/text/"
+      SecRuleUpdateTargetById 942130 "!ARGS:/.*comment/"
```
Comment (petergphillips) [inline helm_deploy/values-cp-dev.yaml:32]: according to mod security docs this is a regular expression so can exclude arguments that end with `comment` in this way (which hopefully matches the `EndsWith` azure WAF syntax)

File: helm_deploy/values-cp-dev.yaml [hunk 4]
```diff
@@ -10,6 +10,28 @@ generic-service:
...
     tlsSecretName: prison-api-cert
+    modsecurity_enabled: true
+    # Missing User Agent Header - 920320
+    # Request Missing an Accept Header - 920300
+    # Restrict file extension - 920440
+    # SQL Injection Attack: Common Injection Testing Detected - 942110
+    # Found User-Agent associated with scripting/generic HTTP client - 913101
+    modsecurity_snippet:
+      SecRuleEngine On
+      SecDefaultAction "phase:2,pass,log,tag:github_team=dps-tech"
```
Comment (petergphillips) [inline helm_deploy/values-cp-dev.yaml:21]: chosen to allow DPS tech members so that others can diagnose Prison API issues too - happy to discuss...

**Review by mikehalmamoj — APPROVED** [review]

File: helm_deploy/values-cp-dev.yaml (hunk 1 above)
Comment (mattops) [reply helm_deploy/values-cp-dev.yaml:13]: Might be an issue with using `toYaml ` in this line https://github.com/ministryofjustice/hmpps-helm-charts/blob/main/charts/generic-service/templates/ingress_cp.yaml#L24

The contents of "modsecurity_snippet" really is not yaml, it's a multiline string.

**Pull request merged**